### PR TITLE
[cli] Search for game executable case-insensitively

### LINKED ISSUE #531: Executable search should be case-insensitive
## Platform
Operating system and version: Arch Linux
Legendary version (`legendary -V`): 0.20.31

## Expected Behavior
Legendary should import the game, searching for the executable case-insensitively. Note that this should probably only happen when on Windows / running a Windows game through Wine

## Current Behavior
When importing a game, Legendary makes sure that the main game executable exists. This is currently done using a simple `os.path.exists`. A problem manifests itself if the executable name set in the manifest does not match the actual executable's case. I've encountered this with Witcher 3, where the executable is named "redprelauncher.exe" in the manifest & "REDprelauncher.exe" on disk

## Steps to Reproduce
1. Install game with this issue (a combination of `legendary info` and `legendary list-files` can probably be used to find more affected games)
2. Uninstall it with `--keep-files`
3. Try to import it again

### PR DESCRIPTION
**Pull request by CommandMC**
Resolves #531 

It should be noted that I have not properly tested `case_insensitive_file_search` on Windows (might be some edge cases with relative paths or something), but then again the function shouldn't be used on Windows in the first place

### CODE REVIEW

File: legendary/cli.py [hunk 1]
```diff
@@ -1287,7 +1287,9 @@ def import_game(self, args):
 
         # get everything needed for import from core, then run additional checks.
         manifest, igame = self.core.import_game(game, args.app_path, platform=args.platform)
-        exe_path = os.path.join(args.app_path, manifest.meta.launch_exe.lstrip('/'))
+        exe_path = os.path.join(args.app_path, manifest.meta.launch_exe.lstrip('/').replace('/', os.sep))
```
Comment (derrod) [inline legendary/cli.py:1290]: Why bother replacing the separator with the OS-specific one? Windows supports both, Linux will only ever use `/`.

File: legendary/lfs/wine_helpers.py [hunk 2]
```diff
@@ -20,6 +20,32 @@ def get_shell_folders(registry, wine_pfx):
...
+        # If the computed directory does not exist, add all remaining parts as-is to at least return a valid path
+        # at the end
+        if not os.path.exists(computed_path):
+            computed_path = os.path.join(computed_path, part)
+            continue
+
+        actual_file_or_dirname = next((
+            x for x in os.listdir(computed_path)
+            if x.lower() == part.lower()
+        ), part)
```
Comment (derrod) [inline legendary/lfs/wine_helpers.py:44]: This should always check for a case-sensitive match first, and only then fall back to case-insenstivie search. Just to be safe in some scenarios where both may exist (which is possible on *nix). `case_insensitive_path_search()` was built that way for this reason.

**Review by derrod — COMMENTED** [review]
Sorry it took so long, life's been busy!

File: legendary/cli.py (hunk 1 above)
Comment (CommandMC) [reply legendary/cli.py:1290]: *I think* at the time I wasn't aware that this could only ever be `/` on non-Windows

**Pull request merged**